[10.0[MIG]exchangelib

AaronHForgeFlow opened the pull request:
It seems to work in our tests for simple calendar events and contacts with single addresses. Pending to tests for more complex cases.

@damdam-s

AaronHForgeFlow committed:
[ADD]gitignore

AaronHForgeFlow committed:
[UPT]mig to exchangelib

damdam-s commented on connector_exchange/data/cron.xml:
```diff
@@ -1,13 +1,11 @@
 
     <record forcecreate="True" id="ir_cron_sync_import_contact_partner" model="ir.cron">
       <field name="name">Connector Exchange - Import Contacts</field>
-      <field eval="False" name="active"/>
```
I would have kept the cron inactive by default and activate it only when needed

AaronHForgeFlow replied:
Yes, that was done just for easy testing. I'll put that back. Thanks

damdam-s commented on connector_exchange/models/calendar_event/exporter.py:
```diff
@@ -104,138 +96,142 @@ def fill_free_busy_status(self, calendar):
-            calendar.is_reminder_set.set(True)
-            calendar.reminder_due_by.set(convert_to_exchange(
+            event.is_reminder_set = True
+            event.reminder_due_by = convert_to_exchange(
```
maybe we should use the `parse_date` method defined below

AaronHForgeFlow replied:
Good point!

damdam-s commented on connector_exchange/models/calendar_event/exporter.py:
```diff
@@ -104,138 +96,142 @@ def fill_free_busy_status(self, calendar):
-                                       ResponseTypeType.Unknown)
+                    attendee.email, event)):
+                att = Attendee(
+                    mailbox=Mailbox(email_address='attendee.email',
```
remove quotes around `attendee.email`

damdam-s reviewed (commented):
just a quick code review. starting tests right now :smile:

damdam-s commented on connector_exchange/models/exchange_backend/common.py:
```diff
@@ -42,15 +33,8 @@ def select_versions(self):
-    certificate_location = fields.Char(
-        required=True,
-        default="/opt/exchange_ws/certificates/")
-    location = fields.Char(
```
I must be blind but how/where do you configure the URL to find the exchange server?

damdam-s replied:
I cannot see something like
```
config = Configuration(server='mail.example.com', credentials=credentials)
account = Account(primary_smtp_address='john@example.com', config=config,
                  autodiscover=False, access_type=IMPERSONATION)
```
found in the documentation of the exchangelib

AaronHForgeFlow replied:
I seems to me it is taken from the domain of the email of the impersonation account. I will recheck, but I was able to connect without providing the URL and the certificate.

damdam-s replied:
ok. my customer does not use external email address to use the connector. I will add an option to disable autodiscover feature and provide a config object instead.

AaronHForgeFlow replied:
You are right. Maybe we can add a boolean field to indicate autodiscover or not.

damdam-s replied:
that's what I was thinking about. I will make a PR against yours :smile:

AaronHForgeFlow committed:
!fixup review round

AaronHForgeFlow committed:
[UPT]requierements

damdam-s commented:
@aheficent I managed to connect via specific URL. 

Now, when exporting a calendar event, I have a 6h difference between Odoo and Exchange. Any idea on this?

AaronHForgeFlow commented:
Good. I think the timezone is hardcoded to East Coast in USA. My fault. I'll fix that.

damdam-s committed:
[ADD] support for not "autodiscover"

[FIX] TZ while exporting calendar event.
dates are given in UTC by Odoo so fix it to UTC

damdam-s committed:
fix delete event in exchange and run import cron in Odoo to delete event in Odoo

damdam-s committed:
manage event attendees and remove import filter on Odoo category

damdam-s committed:
import allday event + fix priority

damdam-s committed:
manage all days event from Odoo to exchange

damdam-s committed:
fix contact import

damdam-s committed:
fix exporter for contacts

damdam-s committed:
change version because of the library change
